### PR TITLE
Add support for the remaining IS expressions.

### LINKED ISSUE #2472: sql: support various IS operators
We have grammar rules for the following, but no semantic actions and no support in evaluation. These should all be straightforward to implement.

```
<expr> IS TRUE
<expr> IS NOT TRUE
<expr> IS FALSE
<expr> IS NOT FALSE
<expr> IS UNKNOWN
<expr> IS NOT UNKNOWN
<expr> IS DISTINCT FROM <expr>
<expr> IS NOT DISTINCT FROM <expr>
<expr> IS OF ( <type-list> )
<expr> IS NOT OF ( <type-list> )
```

### PR DESCRIPTION
**Pull request by petermattis**
Fixes #2472.

### CODE REVIEW

File: sql/parser/eval.go [hunk 1]
```diff
@@ -1116,16 +1119,95 @@ func evalNotExpr(expr *NotExpr) (Datum, error) {
...
+				return result, nil
+			}
+		}
+
+	case DString:
+		for _, t := range expr.Types {
+			if _, ok := t.(*StringType); ok {
+				return result, nil
+			}
+			if _, ok := t.(*BytesType); ok {
```
Comment (bdarnell) [inline sql/parser/eval.go]: We distinguish between string and bytes at the driver level (both in cockroach/sql/driver and in the postgres protocol). Do we really conflate them here? I think we need to maintain this distinction at all levels.

**Comment by bdarnell** [comment]
LGTM

File: sql/parser/eval.go (hunk 1 above)
Comment (petermattis) [inline sql/parser/eval.go]: We conflate them when we fill placeholders. Our handling of bytes vs strings in expressions is inadequate at the moment. We do need to keep them as distinct datum types. It has been on my mental TODO list. I'll file an issue.

**Comment by jusongchen** [comment]
Any difference between  IS UNKNOWN and IS NULL?  If yes, can you give an example?

**Comment by petermattis** [comment]
There is no difference between `IS UNKNOWN` and `IS NULL` in our implementation. In postgres, `<expr> IS UNKNOWN` is only valid if `<expr>` is a boolean expression. Perhaps we should enforce that, but we currently don't.

**Pull request merged**